Frends.MicrosoftSQL.ExecuteProcedure/Frends.MicrosoftSQL.ExecuteProcedure/Definitions/Result.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Frends.MicrosoftSQL.ExecuteProcedure.Definitions;
+
+/// <summary>
+/// Task's result.
+/// </summary>
+public class Result
+{
+    /// <summary>
+    /// Operation complete without errors.
+    /// </summary>
+    /// <example>true</example>
+    public bool Success { get; private set; }
+
+    /// <summary>
+    /// Records affected.
+    /// Some statements will return -1. See Input.ExecuteType's documentation for more information.
+    /// </summary>
+    /// <example>100</example>
+    public int RecordsAffected { get; private set; }
+
+    /// <summary>
+    /// Error message.
+    /// This value is generated when an exception occurs and Options.ThrowErrorOnFailure = false.
+    /// </summary>
+    /// <example>Login failed for user 'user'.</example>
+    public string ErrorMessage { get; private set; }
+
+    /// <summary>
+    /// Procedure's result as JToken.
+    /// </summary>
+    /// <example>
+    /// Input.ExecuteType = ExecuteReader: [{"ID": "1","FIRST_NAME": "Saija","LAST_NAME": "Saijalainen","START_DATE": ""}],
+    /// Input.ExecuteType = NonQuery: {{  "AffectedRows": -1 }},
+    /// Input.ExecuteType = Scalar: {{  "Value": 1 }}
+    /// </example>
+    public dynamic QueryResult { get; private set; }
+
+    internal Result(bool success, int recordsAffected, string errorMessage, dynamic queryResult)
+    {
+        Success = success;
+        RecordsAffected = recordsAffected;
+        ErrorMessage = errorMessage;
+        QueryResult = queryResult;
```

Comment:
Let's rename this to `Data`. In general this would be upcoming naming convention in the future too. Otherwise we keep inventing new property names.

Also, can we fix the ExecuteQuery with the same thing, if it is the same there?

Reply:
Renamed. I'll create new issues for ExecuteQuery and BulkInsert.